sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs

```diff
@@ -126,26 +113,31 @@ public async Task DownloadAsync_ReadFromSecondaryStorage()
                 }
 
                 // Act
-                var response = (Response<BlobDownloadInfo>)await this.EnsurePropagatedAsync(
-                new GetPerformer(this.GetBlob),
-                new HasResponse(this.HasBlob),
-                blockBlobClient);
+                var response = await this.EnsurePropagatedAsync(
+                    async () => await blockBlobClient.DownloadAsync(),
+                    downloadInfo => downloadInfo.GetRawResponse().Status != 404);
 
                 // Assert
                 Assert.AreEqual(data.Length, response.Value.ContentLength);
                 var actual = new MemoryStream();
                 await response.Value.Content.CopyToAsync(actual);
                 TestHelper.AssertSequenceEqual(data, actual.ToArray());
-                Assert.AreEqual(SecondaryStorageTenantPrimaryHost(), testExceptionPolicy.Hosts[0]);
-                Assert.AreEqual(SecondaryStorageTenantSecondaryHost(), testExceptionPolicy.Hosts[1]);
+                Assert.AreEqual(this.SecondaryStorageTenantPrimaryHost(), testExceptionPolicy.HostsSetInRequests[0]);
+                Assert.AreEqual(this.SecondaryStorageTenantSecondaryHost(), testExceptionPolicy.HostsSetInRequests[1]);
             }
         }
 
         [Test]
         public async Task DownloadAsync_ReadFromSecondaryStorageShouldNotPut()
         {
             TestExceptionPolicy testExceptionPolicy;
-            using (this.GetNewContainer(out var container, this.GetServiceClient_SecondaryAccount_ReadEnabledOnRetry(1, out testExceptionPolicy, false, new List<Core.Pipeline.RequestMethod>(new Core.Pipeline.RequestMethod[] { RequestMethod.Put }))))
+            using (this.GetNewContainer(
+                out var container,
+                this.GetServiceClient_SecondaryAccount_ReadEnabledOnRetry(
```

Comment:
Nit - I'd generally pull this out to avoid multiple levels of indentation as part of the same expression